ubidots/ubidots_in.html

```diff
@@ -1,111 +1,608 @@
+
+    oneditsave: function() {},
+    oneditprepare: function() {
+      var variablesActivos = [];
+      var variablesDesactivados = [2, 3, 4, 5, 6, 7, 8, 9, 10];
+      var checkedVariables = [];
+      var auxArray = [];
+      var genericString = "#variable-";
+      var genericCheckBoxString = "#ubidots-checkbox-variable-";
+      var genericLastValueCheckBoxString = "#node-input-checkbox_variable_";
+      var lastValueString = "_last_value";
+
+      //hide/show extra variables by default
+      for (var h = 2; h < 11; h++) {
+        if (!$("#node-input-label_variable_" + h).val()) {
+          $("#variable-" + h).hide();
+          $(
+            "#node-input-checkbox_variable_" + h.toString() + "_last_value"
+          ).prop("checked", true);
+        }
+      }
+
+      if ($("#node-input-custom_topic_checkbox").is(":checked")) {
+        //Custom Topics selected
+        for (var i = 1; i < 11; i++) {
+          $(
+            "#node-input-checkbox_variable_" + i.toString() + "_last_value"
+          ).hide();
+          $("#node-input-last-value-label-" + i.toString()).hide();
+          $("#node-input-label_variable_" + i.toString()).attr(
+            "placeholder",
+            "Topic " + i
+          );
+          $("#node-input-label_variable_" + i.toString()).attr(
+            "style",
+            "width:" + "calc(100% - 115px)"
+          );
+        }
+        $("#ubidots-add-variable-button-label").text("Add Topic");
+      } else {
+        //No Custom Topics --> Variables
+        $("#node-input-variable-1-space").show();
+        for (var i = 1; i < 11; i++) {
+          $(
+            "#node-input-checkbox_variable_" + i.toString() + "_last_value"
+          ).show();
+          $("#node-input-last-value-label-" + i.toString()).show();
+          $("#node-input-label_variable_" + i.toString()).attr(
+            "placeholder",
+            "Variable Label " + i
+          );
+          $("#node-input-label_variable_" + i.toString()).attr(
+            "style",
+            "width:" + "calc(100% - 250px)"
+          );
+        }
+        $("#ubidots-add-variable-button-label").text("Add Variable");
+      }
+
+      $("#delete-variables-button").hide();
+      $("#select-all-variables-button").hide();
+      $("#node-input-custom_topic_checkbox").on("click", function() {
+        //checkbox is checked
+        if ($("#node-input-custom_topic_checkbox").is(":checked")) {
+          for (var i = 1; i < 11; i++) {
+            //Changing Placeholders to "Topic"
+
+            $("#node-input-label_variable_" + i).attr(
+              "placeholder",
+              "Topic " + i
+            );
+            //Changing input size
+            $("#node-input-label_variable_" + i).attr(
+              "style",
+              "width:" + "calc(100% - 115px)"
+            );
+            //hiding last value
+            $(
+              "#node-input-checkbox_variable_" + i.toString() + "_last_value"
```

Comment:
No need to call the `i.toString()` explicitly, JS does this automatically as you're appending that variable to a string (it's called coercion).